From b2f31b8935c50c412f1b45e2ec18ddd3f62ef4d3 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Fri, 21 Feb 2025 15:28:57 +0100
Subject: [PATCH 1626/1695] plat-stm32mp1: cleanup the include of stpmic1 file

Let only the files that need them to include 'stm32mp1_stpmic1.h'
and 'stpmic1.h'.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1091059362f2dd0a561f90499da195882140ab16 plat-stm32mp1: cleanup the include of stpmic1 file]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I5393c66efd8b229998275b86bb3f1f13f71dbd05
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/441057
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/pm/low_power.c | 1 +
 core/arch/arm/plat-stm32mp1/pm/psci.c      | 2 --
 core/arch/arm/plat-stm32mp1/stm32_util.h   | 1 -
 3 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/pm/low_power.c b/core/arch/arm/plat-stm32mp1/pm/low_power.c
index f0b680b4b..1419086cd 100644
--- a/core/arch/arm/plat-stm32mp1/pm/low_power.c
+++ b/core/arch/arm/plat-stm32mp1/pm/low_power.c
@@ -14,6 +14,7 @@
 #include <drivers/stm32mp_dt_bindings.h>
 #include <drivers/stm32mp1_ddrc.h>
 #include <drivers/stm32mp1_pwr.h>
+#include <drivers/stm32mp1_stpmic1.h>
 #ifdef CFG_STM32MP13
 #include <drivers/stm32mp13_rcc.h>
 #else
diff --git a/core/arch/arm/plat-stm32mp1/pm/psci.c b/core/arch/arm/plat-stm32mp1/pm/psci.c
index 5a74a77c5..78cd4f5b4 100644
--- a/core/arch/arm/plat-stm32mp1/pm/psci.c
+++ b/core/arch/arm/plat-stm32mp1/pm/psci.c
@@ -10,9 +10,7 @@
 #include <drivers/clk.h>
 #include <drivers/rstctrl.h>
 #include <drivers/stm32mp_dt_bindings.h>
-#include <drivers/stm32mp1_stpmic1.h>
 #include <drivers/stm32mp1_rcc.h>
-#include <drivers/stpmic1.h>
 #include <initcall.h>
 #include <io.h>
 #include <keep.h>
diff --git a/core/arch/arm/plat-stm32mp1/stm32_util.h b/core/arch/arm/plat-stm32mp1/stm32_util.h
index 81612716e..5b90290c4 100644
--- a/core/arch/arm/plat-stm32mp1/stm32_util.h
+++ b/core/arch/arm/plat-stm32mp1/stm32_util.h
@@ -9,7 +9,6 @@
 #include <assert.h>
 #include <drivers/clk.h>
 #include <drivers/pinctrl.h>
-#include <drivers/stm32mp1_stpmic1.h>
 #include <drivers/stm32mp1_rcc_util.h>
 #include <kernel/panic.h>
 #include <stdbool.h>
-- 
2.47.2

